src/main/cpp/include/config/variant_query_config.h

```diff
@@ -55,6 +55,60 @@ class UnknownQueryAttributeException : public std::exception {
   std::string msg_;
 };
 
+class interval_expander {
+  public:
+    struct interval {
+      // left of interval, right of interval, size of interval, size of all preceeding intervals
+      int64_t first, second, total_size_before;
+      interval(int64_t first = -1, int64_t second = -1, int64_t total_size_before = -1) : first(first), second(second), total_size_before(total_size_before) { }
+      bool operator<(const interval& r) const {
+        return first < r.first;
+      }
+      // number of rows in interval
+      inline int64_t size() const { return second - first + 1; }
+      inline int64_t span() const { return second - first; }
+      // total number of rows in interval and all preceeding
+      inline int64_t size_inclusive() const { return second - first + 1 + total_size_before; }
+      bool intersects(const interval& r) const {
+        return second >= r.first && first <= r.second;
+      }
+      // bounding interval/generalized intersection
+      interval operator+(const interval& r) const {
```

Comment:
Can you write a simple test with intersecting ranges to exercise this code and quiet codecov please?